optimum/bettertransformer/models/encoder_models.py

```diff
@@ -751,3 +751,111 @@ def forward(self, hidden_states, attention_mask, **__):
         if hidden_states.is_nested and self.is_last_layer:
             hidden_states = hidden_states.to_padded_tensor(0.0)
         return (hidden_states,)
+
+
+class FSMTEncoderLayerBetterTransformer(BetterTransformerBaseLayer):
+    def __init__(self, fsmt_layer, config):
+        r"""
+                A simple conversion of the FSMT Encoder layer to its `BetterTransformer` implementation.
+
+                Args:
+                    fsmt_layer (`torch.nn.Module`):
+                        The original FSMT Layer where the weights needs to be retrieved.
+                """
+        super().__init__(config)
+        # In_proj layer
+        self.in_proj_weight = nn.Parameter(
+            torch.cat(
+                [
+                    fsmt_layer.self_attn.self.q_proj.weight,
+                    fsmt_layer.self_attn.self.k_proj.weight,
+                    fsmt_layer.self_attn.self.v_proj.weight,
+                ]
+            )
+        )
+        self.in_proj_bias = nn.Parameter(
+            torch.cat(
+                [
+                    fsmt_layer.self_attn.self.q_proj.bias,
+                    fsmt_layer.self_attn.self.k_proj.bias,
+                    fsmt_layer.self_attn.self.v_proj.bias,
+                ]
+            )
+        )
```

Comment:
```suggestion
        self.in_proj_weight = nn.Parameter(
            torch.cat(
                [
                    fsmt_layer.self_attn.q_proj.weight,
                    fsmt_layer.self_attn.k_proj.weight,
                    fsmt_layer.self_attn.v_proj.weight,
                ]
            )
        )
        self.in_proj_bias = nn.Parameter(
            torch.cat(
                [
                    fsmt_layer.self_attn.q_proj.bias,
                    fsmt_layer.self_attn.k_proj.bias,
                    fsmt_layer.self_attn.v_proj.bias,
                ]
            )
        )
```
This should fix some failing tests